### ISSUE TITLE
Show logs for terminated/crashlooping pods/containers

### ISSUE THREAD

**Issue by aronchick** [kubernetes/kubernetes#7686]
I'm going to use a persistent disk in Google Compute Engine, but this would happen with any stored disk.

Repro steps:
- Use default mysql image from Docker hub
- Create a  GCE persistent disk
`gcloud compute disks create --size=10GB --zone=us-central1-a tmp-disk`
- Attach the drive to a node
`gcloud compute instances attach-disk my-instance-node --disk tmp-disk --zone us-central-1a`
- Mount that drive on a node:
`/usr/share/google/safe_format_and_mount -m "mkfs.ext4 -F" /dev/disk/by-id/google-persistent-disk-1 /data`
- Observe that there is a 'lost+found' folder in that directory:
```
# ls /data/
lost+found
```
- Attempt to start the MySQL container with the drive as a volume (this should fail due to a file being in the directory):
```
# docker run -v /data:/var/lib/mysql -e MYSQL_ROOT_PASSWORD=tempPass mysql
Initializing database
2015-05-03T18:30:56.751559Z 0 [ERROR] --initialize specified but the data directory has files in it. Aborting.
2015-05-03T18:30:56.751665Z 0 [ERROR] Aborting
```
- Now, attempt to do the same in Kubernetes
- Notice that the mysql container just keeps thrashing, and never comes up (again, to be expected)

Expected:
Somewhere, in some log (either in kubectl describe or in the kubelet.log) I would hope that the log of the docker containers are exported by default. Because these containers kept coming up, failing and being destroyed, I had to find the docker log only during the time it was available. Basically, here's an example of what I was seeing:

```
# ~/Code/images [13:07:15]
$ gcloud alpha container kubectl describe pod mysql-g87af
W0501 13:10:57.410898   64322 request.go:288] field selector: v1beta1 - events - involvedObject.uid - 5c80a3cc-f035-11e4-9e1d-42010af08960: need to check if this is versioned correctly.
W0501 13:10:57.411260   64322 request.go:288] field selector: v1beta1 - events - involvedObject.id - mysql-g87af: need to check if this is versioned correctly.
W0501 13:10:57.411265   64322 request.go:288] field selector: v1beta1 - events - involvedObject.namespace - default: need to check if this is versioned correctly.
Name:       mysql-g87af
Image(s):     gcr.io/blue-ocean-10016/mysql_image:latest
Host:       node-prefix-node-1.c.blue-ocean-10016.internal/130.211.159.115
Labels:       name=mysql
Status:       Running
Replication Controllers:  mysql (1/1 replicas created)
Containers:
  mysql:
    Image:    gcr.io/blue-ocean-10016/mysql_image:latest
    State:    Running
      Started:    Fri, 01 May 2015 13:10:53 -0700
    Ready:    False
    Restart Count:  12
Conditions:
  Type    Status
  Ready   False
Events:
  FirstSeen       LastSeen      Count From                        SubobjectPath   Reason  Message
  Fri, 01 May 2015 12:11:01 -0700 Fri, 01 May 2015 12:11:01 -0700 1 {kubelet node-prefix-node-1.c.blue-ocean-10016.internal}  spec.containers{mysql}  created Created with docker id e1b08335ab5ceb7f5f989b43f0df0a2fac78e799220b2d16cf3c125af9b9a7c8
  Fri, 01 May 2015 12:11:01 -0700 Fri, 01 May 2015 12:11:01 -0700 1 {kubelet node-prefix-node-1.c.blue-ocean-10016.internal}  spec.containers{mysql}  started Started with docker id e1b08335ab5ceb7f5f989b43f0df0a2fac78e799220b2d16cf3c125af9b9a7c8
  Fri, 01 May 2015 12:11:11 -0700 Fri, 01 May 2015 12:11:11 -0700 1 {kubelet node-prefix-node-1.c.blue-ocean-10016.internal}  spec.containers{mysql}  created Created with docker id 540ff044e0f3304223f315f7619f31366781afdaf52fddde42c03bb903e722bb

[... Repeat 1000 times ...]

  Fri, 01 May 2015 13:10:43 -0700 Fri, 01 May 2015 13:10:43 -0700 1 {kubelet node-prefix-node-1.c.blue-ocean-10016.internal}  spec.containers{mysql}  created Created with docker id e747df4942511fa654567623a509bbe5db7c8945175c59ddaba8865f55d83e5c
  Fri, 01 May 2015 12:08:13 -0700 Fri, 01 May 2015 13:10:53 -0700 343 {kubelet node-prefix-node-1.c.blue-ocean-10016.internal}  spec.containers{mysql}  pulled  Successfully pulled image "gcr.io/blue-ocean-10016/mysql_image:latest"
  Fri, 01 May 2015 13:10:53 -0700 Fri, 01 May 2015 13:10:53 -0700 1 {kubelet node-prefix-node-1.c.blue-ocean-10016.internal}  spec.containers{mysql}  created Created with docker id 0c4555880497f785affdb4005355c5b93ef1557c1efdf28dee943722fbf1c5e9
  Fri, 01 May 2015 13:10:53 -0700 Fri, 01 May 2015 13:10:53 -0700 1 {kubelet node-prefix-node-1.c.blue-ocean-10016.internal}  spec.containers{mysql}  started Started with docker id 0c4555880497f785affdb4005355c5b93ef1557c1efdf28dee943722fbf1c5e9

root@node-prefix-node-1:/home/this_user# docker ps -a
CONTAINER ID        IMAGE                                             COMMAND                CREATED              STATUS                          PORTS               NAMES
1914b0cf19bb        gcr.io/blue-ocean-10016/mysql_image:latest   "/entrypoint.sh mysq   3 seconds ago        Exited (1) 3 seconds ago                            k8s_mysql.c80d9d0c_mysql-g87af_default_5c80a3cc-f035-11e4-9e1d-42010af08960_c766ae6d
c12b1aa56b1f        gcr.io/blue-ocean-10016/mysql_image:latest   "/entrypoint.sh mysq   13 seconds ago       Exited (1) 13 seconds ago                           k8s_mysql.c80d9d0c_mysql-g87af_default_5c80a3cc-f035-11e4-9e1d-42010af08960_07606447
1ea424644c80        gcr.io/blue-ocean-10016/mysql_image:latest   "/entrypoint.sh mysq   23 seconds ago       Exited (1) 23 seconds ago                           k8s_mysql.c80d9d0c_mysql-g87af_default_5c80a3cc-f035-11e4-9e1d-42010af08960_5ac2c4e7
49477f2ed0c9        gcr.io/blue-ocean-10016/mysql_image:latest   "/entrypoint.sh mysq   33 seconds ago       Exited (1) 33 seconds ago                           k8s_mysql.

[ ... Repeat 1000 times ...]

17ac3715f8e9        gcr.io/blue-ocean-10016/rails_image:latest   "/sbin/my_init"        25 hours ago         Exited (0) About an hour ago                        k8s_rails.10b3851f_rails-ftghd_default_291dc136-ef67-11e4-9e1d-42010af08960_6fced842
cb7a501092de        gcr.io/google_containers/pause:0.8.0              "/pause"               25 hours ago         Exited (0) About an hour ago                        k8s_POD.d41d03ce_rails-ftghd_default_291dc136-ef67-11e4-9e1d-42010af08960_bed821e0
7ecd56bfe6f0        gcr.io/blue-ocean-10016/mysql_image:latest   "/entrypoint.sh mysq   39 hours ago         Exited (0) 37 hours ago                             k8s_mysql.c80d9d0c_mysql-sqhv0_default_3831367a-eef7-11e4-9e1d-42010af08960_cf8d6b22
60538e1938fe        gcr.io/google_containers/pause:0.8.0              "/pause"               39 hours ago         Exited (0) 37 hours ago                             k8s_POD.2dd30668_mysql-sqhv0_default_3831367a-eef7-11e4-9e1d-42010af08960_28ed3dd0
0bf1c1dbf156        gcr.io/google_containers/fluentd-gcp:1.2          "/bin/sh -c '/usr/sb   39 hours ago         Up 39 hours                                         k8s_fluentd-gcp.49e92f51_fluentd-to-gcp-node-prefix-node-1_default_5fdad33dc95769b44c7505737f18676a_b2de0f1d
0f9a82135a00        gcr.io/google_containers/pause:0.8.0              "/pause"               39 hours ago         Up 39 hours                                         k8s_POD.1a1fe303_fluentd-to-gcp-node-prefix-node-1_default_5fdad33dc95769b44c7505737f18676a_28588f73
root@node-prefix-node-1:/home/this_user# docker logs 1914b0cf19bb
Initializing database
2015-05-01T20:12:40.841862Z 0 [ERROR] --initialize specified but the data directory has files in it. Aborting.
2015-05-01T20:12:40.841942Z 0 [ERROR] Aborting
```

Only finally, at the end, was I able to both grab the name of the container, pull the docker log and understand what was going on before it got killed.

**Comment by roberthbailey** [comment]
kubectl has a function to fetch logs for a pod but it doesn't work well when the pod is crash looping. See #4526.

**Comment by roberthbailey** [comment]
I meant to /cc @dchen1107 rather than assign this to her.

**Comment by lavalamp** [comment]
It would be awesome if something got the relevant error message into a termination message in the pod.

**Comment by smarterclayton** [comment]
This was fixed in 1.2
